misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/ExecutionContext.java

```diff
@@ -102,12 +105,20 @@ public BufferedReader getErrorStream() {
         return new BufferedReader(new InputStreamReader(launchedProcess.getErrorStream(), StandardCharsets.UTF_8));
     }
 
-    public DebugInstruction getLastInstruction() {
-        return lastInstruction;
+    public DebugInstruction getPrevInstruction() {
```

Comment:
Since the previous instruction can be null, we could return an Optional here.

Reply:
We can work on adding Optional API support to the debugger execution context as a separate improvement, as it has no performance or functionality concerns. Wdyt?

Reply:
+1